src/wizards/tokenSwapDealWizard/stages/tokenDetailsStage/tokenDetailsStage.ts

```diff
@@ -101,7 +101,7 @@ export class TokenDetailsStage {
 
       name: "",
       symbol: "",
-      decimals: 18,
+      decimals: NaN,
```

Comment:
why not undefined instead of NaN?

Reply:
type defintion of `decimals` is only `number`.
I'll change the type to `number | undefined`.

Reply:
I wonder if it would be best to just work with the TokenService system that sets 0 as the default, consider 0 to be not set, and don't allow it to go undefined when the user clears the field?

Reply:
One problem with that: Cannot distinguish decimal value to disable Amount field

Background: Want `decimals = undefined` to be able to disable the Total amount field
_Relevant quote from Telegram:_
> Does this mean that, on the token details card, we have to have the "Amount" disabled while waiting for the 'Decimals" to load from the internet after the user fills in the "Address"?

> yes, which seems reasonable anyway

With `decimals` defaulting the 0, we can have a case, where
1. User fills in "Total amount" value --> decimal is 0
2. User fills in "Token address" --> decimal changes to eg. 18
3. --> "Total amount" value changes, because `decimal` changed (if my knowledge is correct)